node/rpc/rpc.go

```diff
@@ -0,0 +1,19 @@
+package rpc
+
+import (
+	"github.com/celestiaorg/celestia-node/rpc"
+)
+
+// Config represents the configuration for the RPC client
+// used for retrieving information from Celestia Core nodes // TODO @renaynay: update this post-devnet
+type Config struct {
+	Protocol   string
+	RemoteAddr string
+}
+
+// Components collects all the components and services related to the RPC client.
+func Components(cfg *Config) interface{} {
```

Comment:
Btw, we discussed that we won't return an interface, but a type.

However, specifically for the Components case, let's return `fx.Option`, similar to how it's done in node/p2p. Compare line by line.